### PR TITLE
Shell-quote spawn arguments if needed

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This should globally address issues like https://github.com/appium/appium-adb/pull/315
Although, I'm not quite sure whether spawn command should be quoted as well.

### CODE REVIEW

File: lib/exec.js [hunk 1]
```diff
@@ -27,7 +27,11 @@ function exec (cmd, args = [], opts = {}) {
   return new B((resolve, reject) => {
     // spawn the child process with options; we don't currently expose any of
     // the other 'spawn' options through the API
-    let proc = spawn(cmd, args, {cwd: opts.cwd, env: opts.env, shell: opts.shell});
+    let proc = spawn(cmd, args.map(arg => quoteSpawnArgument(arg)), {
```
Comment (imurchie) [inline lib/exec.js]: Seems like the helper method ought to be written so the mapping doesn't need to be done outside of it every time it's used.

File: lib/utils.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import _ from 'lodash';
+import { quote } from 'shell-quote';
+
+function quoteSpawnArgument (arg) {
+  if (_.isEmpty(arg)) {
+    return arg;
+  }
+
+  if (/\s+/.test(`${arg}`) &&
```
Comment (imurchie) [inline lib/utils.js]: These regular expressions could be clarified a bit.

File: lib/exec.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/exec.js]: I was thinking about applying the same method to the `cmd` argument

Comment (imurchie) [reply lib/exec.js]: Two methods is not the worst thing. Or a method that maps if it's an array, and doesn't if it is not.

Comment (mykola-mokhnach) [reply lib/exec.js]: Implemented the second option

**Comment by mykola-mokhnach** [comment]
This might depend on node version. The tests are failing though

File: lib/utils.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ *
+ * @returns {Array<*>} The args array with quoted items.
+ * Other items are simply left untouched.
+ */
+function quoteSpawnArguments (args = []) {
+  if (_.isEmpty(args)) {
+    return args;
+  }
+  if (!_.isArray(args)) {
+    args = [args];
```
Comment (imurchie) [inline lib/utils.js]: This seems wrong. Passing in a single, non-array, argument will result in an array being returned.

**Pull request closed without merging**